src/games/strategy/triplea/util/TransportUtils.java

```diff
@@ -194,4 +134,70 @@ public int compare(final Unit o1, final Unit o2) {
     return totalLoad;
   }
 
+  public static int getTransportCost(final Collection<Unit> units) {
+    if (units == null) {
+      return 0;
+    }
+    int cost = 0;
+    final Iterator<Unit> iter = units.iterator();
+    while (iter.hasNext()) {
+      final Unit item = iter.next();
+      cost += UnitAttachment.get(item.getType()).getTransportCost();
+    }
+    return cost;
+  }
+
+  /**
+   * Returns a map of unit -> transport. Unit must already be loaded in the transport.
+   */
+  private static Map<Unit, Unit> mapTransportsAlreadyLoaded(final Collection<Unit> units,
+      final Collection<Unit> transports) {
+    final Collection<Unit> canBeTransported = Match.getMatches(units, Matches.UnitCanBeTransported);
+    final Collection<Unit> canTransport = Match.getMatches(transports, Matches.UnitCanTransport);
+    final Map<Unit, Unit> mapping = new HashMap<Unit, Unit>();
+    final Iterator<Unit> land = canBeTransported.iterator();
```

Comment:
The complexity is not trivial in this method. Can you avoid using Iterator? Looks like a foreach or a Java8 stream+filter would both work to simplify things.